commit b0ee303cca52a86292602d60eaa7efdf1b6df436
Author: Stephan Arts <stephan@xfce.org>
Date:   Mon Nov 15 07:10:43 2010 +0100

    Migrate ristretto to libxfce4ui.
     - Thanks to Romain Bouvier <skunnyk@archlinux.fr>, who provided a patch.

diff --git a/configure.in b/configure.in
index 5d59252..ceb59a4 100644
--- a/configure.in
+++ b/configure.in
@@ -52,16 +52,16 @@ dnl Check for i18n support
 XDT_I18N([@LINGUAS@])
 
 XDT_CHECK_PACKAGE([LIBEXIF], [libexif], [0.6.0])
-XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.12.0])
-XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [2.16.0])
-XDT_CHECK_PACKAGE([GTHREAD], [gthread-2.0], [2.16.0])
-XDT_CHECK_PACKAGE([GOBJECT], [gobject-2.0], [2.16.0])
-XDT_CHECK_PACKAGE([GIO], [gio-2.0], [2.16.0])
+XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.14.0])
+XDT_CHECK_PACKAGE([GLIB], [glib-2.0], [2.18.0])
+XDT_CHECK_PACKAGE([GTHREAD], [gthread-2.0], [2.18.0])
+XDT_CHECK_PACKAGE([GOBJECT], [gobject-2.0], [2.18.0])
+XDT_CHECK_PACKAGE([GIO], [gio-2.0], [2.18.0])
 XDT_CHECK_PACKAGE([DBUS_GLIB], [dbus-glib-1], [0.34])
-XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [4.6.0])
-XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.6.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [4.7.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.7.0])
 
-XDT_CHECK_PACKAGE([XFCONF], [libxfconf-0], [4.6.0])
+XDT_CHECK_PACKAGE([XFCONF], [libxfconf-0], [4.7.0])
 
 XDT_CHECK_PACKAGE([CAIRO], [cairo], [1.8.0])
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 82c6bf7..dd1fc5a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -29,7 +29,7 @@ ristretto_CFLAGS = \
 	$(XFCONF_CFLAGS) \
 	$(CAIRO_CFLAGS) \
 	$(LIBXFCE4UTIL_CFLAGS) \
-	$(LIBXFCEGUI4_CFLAGS) \
+	$(LIBXFCE4UI_CFLAGS) \
 	-DDATADIR=\"$(datadir)\" \
 	-DSRCDIR=\"$(top_srcdir)\" \
 	-DLOCALEDIR=\"$(localedir)\"
@@ -43,7 +43,7 @@ ristretto_LDADD = \
 	$(CAIRO_LIBS) \
 	$(XFCONF_LIBS) \
 	$(LIBXFCE4UTIL_LIBS) \
-	$(LIBXFCEGUI4_LIBS)
+	$(LIBXFCE4UI_LIBS) -lX11
 
 INCLUDES = \
 	-I${top_srcdir}
diff --git a/src/main_window.c b/src/main_window.c
index 2e61084..029ca7c 100644
--- a/src/main_window.c
+++ b/src/main_window.c
@@ -12,9 +12,7 @@
  *  GNU Library General Public License for more details.
  *
  *  You should have received a copy of the GNU General Public License
- *  along with this program; if not, write to the Free Software
- *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
- */
+ *  along with this program; if not, write to the Free Software *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */ 
 
 #include <config.h>
 #include <gtk/gtk.h>
@@ -24,7 +22,7 @@
 
 #include <gio/gio.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <libexif/exif-data.h>
 
 #include <cairo/cairo.h>
diff --git a/src/preferences_dialog.c b/src/preferences_dialog.c
index fde5635..738f005 100644
--- a/src/preferences_dialog.c
+++ b/src/preferences_dialog.c
@@ -18,7 +18,7 @@
 
 #include <config.h>
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <libxfce4util/libxfce4util.h>
 
 #include "settings.h"
@@ -224,7 +224,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
 
 /** Bg-color frame */
     dialog->priv->display_tab.bgcolor_vbox = gtk_vbox_new (FALSE, 0);
-    dialog->priv->display_tab.bgcolor_frame = xfce_create_framebox_with_content (_("Background color"),
+    dialog->priv->display_tab.bgcolor_frame = xfce_gtk_frame_box_new_with_content(_("Background color"),
                                                                                  dialog->priv->display_tab.bgcolor_vbox);
     gtk_box_pack_start (GTK_BOX (display_main_vbox), dialog->priv->display_tab.bgcolor_frame, FALSE, FALSE, 0);
 
@@ -269,7 +269,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
     gtk_notebook_append_page(GTK_NOTEBOOK(notebook), slideshow_main_vbox, slideshow_main_lbl);
 
     dialog->priv->slideshow_tab.timeout_vbox = gtk_vbox_new(FALSE, 0);
-    dialog->priv->slideshow_tab.timeout_frame = xfce_create_framebox_with_content (_("Timeout"), dialog->priv->slideshow_tab.timeout_vbox);
+    dialog->priv->slideshow_tab.timeout_frame = xfce_gtk_frame_box_new_with_content(_("Timeout"), dialog->priv->slideshow_tab.timeout_vbox);
     gtk_box_pack_start (GTK_BOX (slideshow_main_vbox), dialog->priv->slideshow_tab.timeout_frame, FALSE, FALSE, 0);
 
     timeout_lbl = gtk_label_new(_("The time period an individual image is displayed during a slideshow\n(in seconds)"));
@@ -287,7 +287,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
 
     
     dialog->priv->slideshow_tab.thumbnail_vbox = gtk_vbox_new(FALSE, 0);
-    dialog->priv->slideshow_tab.thumbnail_frame = xfce_create_framebox_with_content (_("Thumbnails"), dialog->priv->slideshow_tab.thumbnail_vbox);
+    dialog->priv->slideshow_tab.thumbnail_frame = xfce_gtk_frame_box_new_with_content(_("Thumbnails"), dialog->priv->slideshow_tab.thumbnail_vbox);
     gtk_box_pack_start (GTK_BOX (slideshow_main_vbox), dialog->priv->slideshow_tab.thumbnail_frame, FALSE, FALSE, 0);
 
     dialog->priv->slideshow_tab.hide_thumbnails_fullscreen_lbl = gtk_label_new(_("The thumbnailbar can be automatically hidden \nwhen the image-viewer is fullscreen."));
@@ -309,7 +309,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
     gtk_notebook_append_page(GTK_NOTEBOOK(notebook), control_main_vbox, control_main_lbl);
 
     dialog->priv->control_tab.scroll_vbox = gtk_vbox_new(FALSE, 0);
-    dialog->priv->control_tab.scroll_frame = xfce_create_framebox_with_content (_("Scrollwheel"), dialog->priv->control_tab.scroll_vbox);
+    dialog->priv->control_tab.scroll_frame = xfce_gtk_frame_box_new_with_content(_("Scrollwheel"), dialog->priv->control_tab.scroll_vbox);
     gtk_box_pack_start (GTK_BOX (control_main_vbox), dialog->priv->control_tab.scroll_frame, FALSE, FALSE, 0);
 
     dialog->priv->control_tab.no_scrollwheel_action_radio_button = gtk_radio_button_new_with_label (NULL, _("No action"));
@@ -358,7 +358,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
 
     /********************************************/
     dialog->priv->behaviour_tab.scaling_vbox = gtk_vbox_new(FALSE, 0);
-    dialog->priv->behaviour_tab.scaling_frame = xfce_create_framebox_with_content (_("Scaling"), dialog->priv->behaviour_tab.scaling_vbox);
+    dialog->priv->behaviour_tab.scaling_frame = xfce_gtk_frame_box_new_with_content(_("Scaling"), dialog->priv->behaviour_tab.scaling_vbox);
     gtk_box_pack_start (GTK_BOX (behaviour_main_vbox), dialog->priv->behaviour_tab.scaling_frame, FALSE, FALSE, 0);
     /* not used */
     gtk_widget_set_sensitive (dialog->priv->behaviour_tab.scaling_vbox, FALSE);
@@ -367,7 +367,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
     gtk_container_add (GTK_CONTAINER (dialog->priv->behaviour_tab.scaling_vbox), dialog->priv->behaviour_tab.resize_image_on_maximize);
 
     dialog->priv->behaviour_tab.startup_vbox = gtk_vbox_new(FALSE, 0);
-    dialog->priv->behaviour_tab.startup_frame = xfce_create_framebox_with_content (_("Startup"), dialog->priv->behaviour_tab.startup_vbox);
+    dialog->priv->behaviour_tab.startup_frame = xfce_gtk_frame_box_new_with_content(_("Startup"), dialog->priv->behaviour_tab.startup_vbox);
     gtk_box_pack_start (GTK_BOX (behaviour_main_vbox), dialog->priv->behaviour_tab.startup_frame, FALSE, FALSE, 0);
     dialog->priv->behaviour_tab.resize_window_on_startup_check_button = gtk_check_button_new_with_label (_("Resize window to image on startup"));
     gtk_container_add (GTK_CONTAINER (dialog->priv->behaviour_tab.startup_vbox), dialog->priv->behaviour_tab.resize_window_on_startup_check_button);
@@ -400,7 +400,7 @@ rstto_preferences_dialog_init(RsttoPreferencesDialog *dialog)
 
 /** Image-cache frame */
     dialog->priv->cache_tab.cache_vbox = gtk_vbox_new(FALSE, 0);
-    dialog->priv->cache_tab.cache_frame = xfce_create_framebox_with_content (_("Image cache"),
+    dialog->priv->cache_tab.cache_frame = xfce_gtk_frame_box_new_with_content(_("Image cache"),
                                                                                  dialog->priv->cache_tab.cache_vbox);
     gtk_box_pack_start (GTK_BOX (cache_main_vbox), dialog->priv->cache_tab.cache_frame, FALSE, FALSE, 0);
 
diff --git a/src/thumbnail_bar.c b/src/thumbnail_bar.c
index 282db14..e2bc892 100644
--- a/src/thumbnail_bar.c
+++ b/src/thumbnail_bar.c
@@ -21,7 +21,7 @@
 
 #include <gio/gio.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <libexif/exif-data.h>
 
 #include "image.h"
